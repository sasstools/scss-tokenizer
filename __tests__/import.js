var scss = require('..');
var fs = require('fs');
var path = require('path');

var fixture = function(name) {
    return fs.readFileSync(
        path.join(__dirname, 'fixture', name)
    );
}

describe('@import', function() {
    it('should tokenize a @import with double quotes', function() {
        const tree = scss.tokenize(fixture('double-quoted-import.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a @import with single quotes', function() {
        const tree = scss.tokenize(fixture('single-quoted-import.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a @import without quotes', function() {
        const tree = scss.tokenize(fixture('unquoted-import.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a @import with double quotes and slash', function() {
        const tree = scss.tokenize(fixture('double-quoted-import-path.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a @import with single quotes and slash', function() {
        const tree = scss.tokenize(fixture('single-quoted-import-path.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a @import without quotes and slash', function() {
        const tree = scss.tokenize(fixture('unquoted-import-path.scss'));
        expect(tree).toMatchSnapshot();
    });
});
