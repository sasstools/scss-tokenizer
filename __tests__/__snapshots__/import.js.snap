// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@import should tokenize a @import with double quotes 1`] = `
Array [
  Array [
    "@",
    "@",
    1,
    1,
  ],
  Array [
    "ident",
    "import",
    1,
    2,
    1,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "\\"",
    "\\"",
    1,
    9,
  ],
  Array [
    "string",
    "foo",
    1,
    10,
    1,
    12,
  ],
  Array [
    "\\"",
    "\\"",
    1,
    13,
  ],
  Array [
    ";",
    ";",
    1,
    14,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
]
`;

exports[`@import should tokenize a @import with double quotes and slash 1`] = `
Array [
  Array [
    "@",
    "@",
    1,
    1,
  ],
  Array [
    "ident",
    "import",
    1,
    2,
    1,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "\\"",
    "\\"",
    1,
    9,
  ],
  Array [
    "string",
    "foo/bar",
    1,
    10,
    1,
    16,
  ],
  Array [
    "\\"",
    "\\"",
    1,
    17,
  ],
  Array [
    ";",
    ";",
    1,
    18,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
]
`;

exports[`@import should tokenize a @import with single quotes 1`] = `
Array [
  Array [
    "@",
    "@",
    1,
    1,
  ],
  Array [
    "ident",
    "import",
    1,
    2,
    1,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "'",
    "'",
    1,
    9,
  ],
  Array [
    "string",
    "foo",
    1,
    10,
    1,
    12,
  ],
  Array [
    "'",
    "'",
    1,
    13,
  ],
  Array [
    ";",
    ";",
    1,
    14,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
]
`;

exports[`@import should tokenize a @import with single quotes and slash 1`] = `
Array [
  Array [
    "@",
    "@",
    1,
    1,
  ],
  Array [
    "ident",
    "import",
    1,
    2,
    1,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "'",
    "'",
    1,
    9,
  ],
  Array [
    "string",
    "foo/bar",
    1,
    10,
    1,
    16,
  ],
  Array [
    "'",
    "'",
    1,
    17,
  ],
  Array [
    ";",
    ";",
    1,
    18,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
]
`;

exports[`@import should tokenize a @import without quotes 1`] = `
Array [
  Array [
    "@",
    "@",
    1,
    1,
  ],
  Array [
    "ident",
    "import",
    1,
    2,
    1,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "ident",
    "foo",
    1,
    9,
    1,
    11,
  ],
  Array [
    ";",
    ";",
    1,
    12,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
]
`;

exports[`@import should tokenize a @import without quotes and slash 1`] = `
Array [
  Array [
    "@",
    "@",
    1,
    1,
  ],
  Array [
    "ident",
    "import",
    1,
    2,
    1,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "ident",
    "foo",
    1,
    9,
    1,
    11,
  ],
  Array [
    "/",
    "/",
    1,
    12,
  ],
  Array [
    "ident",
    "bar",
    1,
    13,
    1,
    15,
  ],
  Array [
    ";",
    ";",
    1,
    16,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
]
`;
